generated/containerservice/microsoft.containerservice/2022-06-01/types.md

```diff
@@ -221,7 +221,7 @@
 * **enableNodePublicIP**: bool: Some scenarios may require nodes in a node pool to receive their own dedicated public IP addresses. A common scenario is for gaming workloads, where a console needs to make a direct connection to a cloud virtual machine to minimize hops. For more information see [assigning a public IP per node](https://docs.microsoft.com/azure/aks/use-multiple-node-pools#assign-a-public-ip-per-node-for-your-node-pools). The default is false.
 * **enableUltraSSD**: bool: Whether to enable UltraSSD
 * **gpuInstanceProfile**: 'MIG1g' | 'MIG2g' | 'MIG3g' | 'MIG4g' | 'MIG7g' | string: GPUInstanceProfile to be used to specify GPU MIG instance profile for supported GPU VM SKU.
-* **hostGroupID**: string: This is of the form: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/hostGroups/{hostGroupName}. For more information see [Azure dedicated hosts](https://docs.microsoft.com/azure/virtual-machines/dedicated-hosts).
+* **hostGroupID**: any: This is of the form: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/hostGroups/{hostGroupName}. For more information see [Azure dedicated hosts](https://docs.microsoft.com/azure/virtual-machines/dedicated-hosts).
```

Comment:
I previously closed this PR to prevent automatic merging because this change was unexpected. `hostGroupId` is [tagged in the Swagger as an ARM ID](https://github.com/Azure/azure-rest-api-specs/blob/1b3b9c1dd4d2c875997ea0b392dc71418fb1f28d/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json#L2823-L2835), and AutoRest translates this into [its new ARM ID type](https://github.com/Azure/autorest/pull/4611). I will open a follow-up PR to change this back to `string` and see if there's anything more we can do with these.